pkg/backends/datadog/datadog.go

```diff
@@ -258,28 +258,44 @@ func NewClientFromViper(v *viper.Viper) (gostatsd.Backend, error) {
 	dd.SetDefault("api_endpoint", apiURL)
 	dd.SetDefault("metrics_per_batch", defaultMetricsPerBatch)
 	dd.SetDefault("compress_payload", true)
-	dd.SetDefault("dual_stack", false)
+	dd.SetDefault("ipv4", true)
+	dd.SetDefault("ipv6", true)
 	dd.SetDefault("client_timeout", defaultClientTimeout)
 	dd.SetDefault("max_request_elapsed_time", defaultMaxRequestElapsedTime)
+
+	ipv4 := dd.GetBool("ipv4")
+	ipv6 := dd.GetBool("ipv6")
+	var network string
+	if ipv4 && ipv6 {
+		network = "tcp"
+	} else if ipv4 {
+		network = "tcp4"
+	} else if ipv6 {
+		network = "tcp6"
+	}
+
 	return NewClient(
 		dd.GetString("api_endpoint"),
 		dd.GetString("api_key"),
+		network,
 		uint(dd.GetInt("metrics_per_batch")),
 		dd.GetBool("compress_payload"),
-		dd.GetBool("dual_stack"),
 		dd.GetDuration("client_timeout"),
 		dd.GetDuration("max_request_elapsed_time"),
 	)
 }
 
 // NewClient returns a new Datadog API client.
-func NewClient(apiEndpoint, apiKey string, metricsPerBatch uint, compressPayload, dualStack bool, clientTimeout, maxRequestElapsedTime time.Duration) (*Client, error) {
+func NewClient(apiEndpoint, apiKey, network string, metricsPerBatch uint, compressPayload bool, clientTimeout, maxRequestElapsedTime time.Duration) (*Client, error) {
 	if apiEndpoint == "" {
 		return nil, fmt.Errorf("[%s] apiEndpoint is required", BackendName)
 	}
 	if apiKey == "" {
 		return nil, fmt.Errorf("[%s] apiKey is required", BackendName)
 	}
+	if network == "" {
+		return nil, fmt.Errorf("[%s] at least one of ipv4 or ipv6 must be enabled", BackendName)
```

Comment:
Huh, this is the error that should be returned from `NewClientFromViper()`.
And this function does not know anything about ipv4/ipv6 flags. It should error out with "network must not be empty". WDYT?

Reply:
Not for this PR: we should remove backend name from error string and instead add it up the call stack.

Reply:
I think enforcing 4/6 at all is not our problem, and we can pass it off to anything Go supports :)

Reply:
Regarding adding it up the callstack: it should probably be on the logger...  The one we add when we do #2